docs/source/app-dev/app-arch.rst

```diff
@@ -148,6 +148,24 @@ To use command deduplication, you should:
 
 For more details on command deduplication, see the :ref:`Ledger API Services <command-submission-service-deduplication>` documentation.
 
+
+.. _handling-participant-node-failover:
+
+Handling participant node failover
+**********************************
+
+Some DAML Ledgers support exposing multiple eventually consistent Ledger API endpoints hosted by separate participant nodes.
+
+To support switching to a different participant node at run time,
+your application should keep track of the last ledger offset received from the :ref:`transaction service <transaction-service>`.
+After switching to a new participant node, subscribe to the transaction stream using the last received offset.
+If you receive the OUT_OF_RANGE error, it means the new participant node hasn't caught up with the state on the previous participant yet
+and your application should retry subscribing to the transaction stream until successful.
+
+Note that command deduplication is not guaranteed to work across participants.
```

Comment:
Let's change that into an assumption on the eventually consistent ledger api endpoints. Make it explicit that these do perform command-deduplication across each other.

Canton and Corda for example would do it by hosting the IndexDB in a highly-available manner and ensuring that the deduplication info is replicated as well.

Reply:
Last time I checked, there was no defined story for "duplicate participants" for Canton. I think making an assumption about a highly-available IndexDB is premature.

Reply:
Agreed. I wasn't proposing that we add any commentary on this here. I was highlighting though that Canton and Corda could expose such eventually consistent Ledger API endpoints, which is why I believe using such Ledger API endpoints as the precondition for having applications that handle fail-over is OK.

Reply:
@gerolf-da : resolving given our direct conversations. Please reopen if you disagree.